version: "3.3"
services:
  traefik:
    image: "traefik:latest"
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./traefik/acme.json:/acme.json
      - ./traefik/traefik.yml:/etc/traefik/traefik.yml
      - ./traefik/dynamic/:/etc/traefik/dynamic/
    restart: unless-stopped

  grafana:
    image: grafana/grafana:latest
    restart: unless-stopped
    volumes:
      - ./grafana/:/var/lib/grafana
      - ./grafana.ini:/etc/grafana/grafana.ini
    ports:
      - 3000:3000
    env_file: .env
    user: "1000"
    labels:
      - traefik.enable=true
      - traefik.http.routers.grafana.rule=Host(`grafana.datagnome.de`)
      - traefik.http.routers.grafana.entrypoints=websecure
      - traefik.http.routers.grafana.tls=true
      - traefik.http.routers.grafana.tls.certresolver=le
      - traefik.http.services.grafana.loadbalancer.server.port=3000

  influxdb:
    image: influxdb:2.6-alpine
    volumes:
      - ./influx-data:/var/lib/influxdb2:rw
    env_file: .env
    ports:
      - "8086:8086"
    labels:
      - traefik.enable=true
      - traefik.http.routers.influxdb.rule=Host(`influxdb.datagnome.de`)
      - traefik.http.routers.influxdb.entrypoints=websecure
      - traefik.http.routers.influxdb.tls=true
      - traefik.http.routers.influxdb.tls.certresolver=le
      - traefik.http.services.influxdb.loadbalancer.server.port=8086
